src/diffusers/pipelines/controlnet/pipeline_controlnet.py

```diff
@@ -837,6 +843,20 @@ def __call__(
                 second element is a list of `bool`s indicating whether the corresponding generated image contains
                 "not-safe-for-work" (nsfw) content.
         """
+        # We do this first to capture the "true" call values. If we do this at a later point in time,
+        # we cannot ensure that the true call values weren't changed during the process.
+        # We update the `generator` later, though as we define a new generator in case it was passed as `None`.
+        if generator is None:
+            seed = random.randint(0, MAX_SEED)
+            generator = torch.manual_seed(seed)
+
+        workflow = None
+        if return_workflow:
```

Comment:
```suggestion
        if return_workflow:
                    # We do this first to capture the "true" call values. If we do this at a later point in time,
        # we cannot ensure that the true call values weren't changed during the process.
        # We update the `generator` later, though as we define a new generator in case it was passed as `None`.
        if generator is None:
            seed = random.randint(0, MAX_SEED)
            generator = torch.Generator(self.device).manual_seed(seed)
```

Reply:
I think generator initialization should be done regardless of the status of `return_workflow`.

Reply:
Why? If no workflow is returned then we don't have to know the generator no?